utils/hasher/b64hasher.go

```diff
@@ -0,0 +1,53 @@
+package b64hasher
+
+import (
+	"crypto/sha512"
+	"encoding/base64"
+	"math/rand"
+)
+
```

Comment:
I suggest:
1. To Create struct `PasswordJsonEncoder` that have fields salt (because salt could be encapsulated, it is using in every method) and hasher (sha512 implements `Hash` interface, therefore we should use that interface as minimal type to use different hashing func)
2. GenerateSalt should be internal struct func and probably should have the name `GenerateRandomSalt`